lib/saluki-components/src/transforms/dogstatsd_mapper/mod.rs

```diff
@@ -0,0 +1,254 @@
+        pattern = pattern.replace("*", "([^.]*)");
+    }
+
+    let final_pattern = format!("^{}$", pattern);
+
+    match Regex::new(&final_pattern) {
+        Ok(re) => Ok(re),
+        Err(e) => Err(generic_error!(
+            "invalid match `{}`, cannot compile regex: {}",
+            match_re,
+            e
+        )),
+    }
+}
+
+impl MapperProfileConfigs {
+    fn build(&self) -> Result<MetricMapper, GenericError> {
+        let mut profiles = Vec::with_capacity(self.0.len());
+        for (i, config_profile) in self.0.iter().enumerate() {
+            if config_profile.name.is_empty() {
+                return Err(generic_error!("missing profile name"));
+            }
+            if config_profile.prefix.is_empty() {
+                return Err(generic_error!("missing prefix for profile: {}", config_profile.name));
+            }
+
+            let mut profile = MappingProfile {
+                prefix: config_profile.prefix.clone(),
+                mappings: Vec::with_capacity(config_profile.mappings.len()),
+            };
+
+            for mapping in &config_profile.mappings {
+                let mut match_type = mapping.match_type.to_string();
+                if mapping.match_type == "" {
+                    match_type = "wildcard".to_string();
+                }
+                if match_type != MATCH_TYPE_WILDCARD && match_type != MATCH_TYPE_REGEX {
+                    return Err(generic_error!(
+                        "profile: {}, mapping num {}: invalid match type, must be `wildcard` or `regex`",
+                        config_profile.name,
+                        i
+                    ));
+                }
+                if mapping.name == "" {
+                    return Err(generic_error!(
+                        "profile: {}, mapping num {}: name is required",
+                        config_profile.name,
+                        i
+                    ));
+                }
+                if mapping.name == "" {
+                    return Err(generic_error!(
+                        "profile: {}, mapping num {}: match is required",
+                        config_profile.name,
+                        i
+                    ));
+                }
+                let regex = build_regex(&mapping.metric_match, &match_type)?;
+                profile.mappings.push(MetricMapping {
+                    name: mapping.name.clone(),
+                    tags: mapping.tags.clone(),
+                    regex,
+                });
+            }
+            profiles.push(profile);
+        }
+
+        Ok(MetricMapper { profiles })
+    }
+}
+
+impl DogstatsDMapperConfiguration {
+    /// Creates a new `DogstatsDMapperConfiguration` from the given configuration.
+    pub fn from_configuration(config: &GenericConfiguration) -> Result<Self, GenericError> {
+        let mut config: Self = config.as_typed()?;
+        config.context_string_interner_bytes = ByteSize::kib(512);
```

Comment:
I copied the value from the `AgentTelemetryRemapper` as I'm unsure how to determine what value to set this as.